api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -452,6 +454,59 @@ private static void checkSchemaCompatibility(
     }
   }
 
+  public static long defaultSize(Types.NestedField field) {
```

Comment:
I think we have 2 options to limit the size of the cache:
- Add ways to limit the size of equality and position delete files to be loaded.
- Add ways to indicate the maximum cache size in bytes.

The first one is easy but we need to account that delete files may store extra columns and it will be hard to configure these options correctly cause users have no idea how much the cached representation will actually occupy. That's why I went with the estimation approach and let users configure the max cache size in bytes.

Reply:
Why do we need to cache the extra columns? We wouldn't be using them?

Reply:
The idea is to ask the user for the total size of the cache in memory and estimate the actual size to honor those values. For equality deletes, we will rely on the number of records and types of equality columns.